### PR TITLE
Add event name to System.Runtime.Notify()

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo/SmartContract/ApplicationEngine.Runtime.cs [hunk 1]
```diff
@@ -146,15 +147,16 @@ internal void RuntimeLog(byte[] state)
             Log?.Invoke(this, new LogEventArgs(ScriptContainer, CurrentScriptHash, message));
         }
 
-        internal void RuntimeNotify(StackItem state)
+        internal void RuntimeNotify(byte[] eventName, Array state)
         {
+            if (eventName.Length > MaxEventName) throw new ArgumentException();
             if (!CheckItemForNotification(state)) throw new ArgumentException();
-            SendNotification(CurrentScriptHash, state);
+            SendNotification(CurrentScriptHash, Encoding.ASCII.GetString(eventName), state);
```
Comment (shargon) [inline src/neo/SmartContract/ApplicationEngine.Runtime.cs:154]: UTF8?

Comment (erikzhang) [reply src/neo/SmartContract/ApplicationEngine.Runtime.cs:154]: Allow UTF8 in event name? I don't know...

Comment (shargon) [reply src/neo/SmartContract/ApplicationEngine.Runtime.cs:154]: Me neither, but if someone want to use a Chinese characters, it won't be able to do that.

Comment (erikzhang) [reply src/neo/SmartContract/ApplicationEngine.Runtime.cs:154]:
```suggestion
            SendNotification(CurrentScriptHash, Encoding.UTF8.GetString(eventName), state);
```

Comment (erikzhang) [reply src/neo/SmartContract/ApplicationEngine.Runtime.cs]: Modified.

**Pull request merged**